tests/integration/run.bats

```diff
@@ -39,125 +44,185 @@ function docker_node_nabla_run() {
 	echo "$output" >&2
 }
 
-@test "test hello" {
-	nabla_run -unikernel test_hello.nabla
-	[[ "$output" == *"Hello, World"* ]]
-	[ "$status" -eq 0 ]
-}
+function runnc_run() {
+    # trick: don't start with 'run', or you will get a deadlock
+    # waiting for the nabla program to be done but no one start it
+    runnc create "$1" > $RUNNC_OUT
+    run runnc start "$1"
 
-@test "test node-hello from a pre-built node_tests.iso" {
-	nabla_run -unikernel node.nabla -volume node_tests.iso:/hello -- /hello/app.js
-	[[ "$output" == *"hello from node"* ]]
-	[ "$status" -eq 0 ]
+    echo "nabla-run $@ (status=$status):" >&2
+    echo "$output" >&2
+
+    # waiting for nabla log to be done
+    sleep 1
 }
 
-@test "test node-hello runnc" {
-	local-test
+@test "hello" {
+    local name="test-nabla-hello"
 
-	run sudo docker run --rm --runtime=runnc nablact/nabla-node:test /hello/app.js
-	[[ "$output" == *"hello from node"* ]]
-	[ "$status" -eq 0 ]
-}
+	cd "$HELLO_BUNDLE"
+	cat config-hello.json > config.json
 
-@test "test node-env" {
-	nabla_run -unikernel node.nabla -env BLA=bla -env NABLA_ENV_TEST=blableblibloblu -env BLE=ble -volume node_tests.iso:/hello -- /hello/env.js
-	echo "$output"
-	[[ "$output" == *"env=blableblibloblu"* ]]
-	[ "$status" -eq 0 ]
+	runnc_run "$name"
+
+	run cat "$HELLO_BUNDLE/$RUNNC_OUT"
+    [[ "$output" == *"Hello, World"* ]]
+
+	runnc delete --force "$name"
 }
 
-@test "test node-cwd" {
-	nabla_run -unikernel node.nabla -cwd /hello -volume node_tests.iso:/hello -- /hello/cwd.js
-	echo "$output"
-	[[ "$output" == *"cwd=/hello"* ]]
-	[ "$status" -eq 0 ]
+@test "hello with arg" {
+   local name="test-nabla-hello-arg"
+
+	cd "$HELLO_BUNDLE"
+	cat config-hello.json \
+	| jq '.process.args |= .+ ["hola"]' \
+	> config.json
```

Comment:
Seem like this pattern is recurring, how about making it a func?

Reply:
More generically, i think we can use the same `runnc` template for all of them and make a function like

`create_config (arg_array)`

Since you're already doing all the jq parsing :)

What do you think?

Reply:
> More generically, i think we can use the same `runnc` template for all of them and make a function like
> 
> `create_config (arg_array)`
> 
> Since you're already doing all the jq parsing :)
> 
> What do you think?

agree with you😀

Reply:
made it like this instead, since passing in bash is actually messier:
```
    $PRE_CONFIG_MOD_CMD \
    | jq '.process.args |= .+ ["node.nabla", "/hello/app.js"]' \
    $POST_CONFIG_MOD_CMD
```

Reply:
> made it like this instead, since passing in bash is actually messier:
> 
> ```
>     $PRE_CONFIG_MOD_CMD \
>     | jq '.process.args |= .+ ["node.nabla", "/hello/app.js"]' \
>     $POST_CONFIG_MOD_CMD
> ```

LGTM

Reply:
actually, even that's proving difficult with bash... going to try something else.

Reply:
Yea bats does something weird that doesn't make that possible, so i ended up taking making a function that reparses the file each time, but it's a lot cleaner:
```
    config_mod '.process.args |= .+ ["node.nabla", "/hello/cwd.js"]'
    config_mod '.process.cwd |= "/hello"'
```

I will send you a review soon with all of the changes! Should be done by today if nothing weird comes up!